src/activity_actor.cpp

```diff
@@ -4077,6 +4078,127 @@ std::unique_ptr<activity_actor> haircut_activity_actor::deserialize( JsonIn & )
     return haircut_activity_actor().clone();
 }
 
+rummage_pocket_activity_actor::rummage_pocket_activity_actor( const item_location &iloc,
+        const action act_kind, const tripoint &tgt )
+    : kind( act_kind ), target( tgt )
+{
+    std::pair<item_location, int> it_loc = std::make_pair( iloc, iloc->count() );
+    item_loc.emplace_back( it_loc );
+}
+
+void rummage_pocket_activity_actor::start( player_activity &act, Character &who )
+{
+
+    int moves{};
+
+    for( auto item : item_loc ) {
+        moves += item.first.obtain_cost( who );
+    }
+    act.moves_total = moves;
+    act.moves_left = moves;
+}
+
+void rummage_pocket_activity_actor::do_turn( player_activity &, Character &who )
+{
+    who.add_msg_if_player( _( "You rummage your pockets" ) );
```

Comment:
This seems very spammy, I am not sure if there is a point in having it.
```suggestion
    who.add_msg_if_player( _( "You rummage through your pockets" ) );
```

Reply:
I'm relying on that to test if my changes are working but it can be removed after I've finished